src/event/event_queue.rs

```diff
@@ -64,30 +63,51 @@ impl EventQueue {
     /// Creates an unbounded queue for scheduling closures on the JavaScript
     /// main thread
     pub fn new<'a, C: Context<'a>>(cx: &mut C) -> Self {
-        let tsfn = unsafe { ThreadsafeFunction::new(cx.env().to_raw(), Self::callback) };
+        let trampoline = Arc::new(RwLock::new(ThreadsafeTrampoline::new(cx.env().to_raw())));
 
-        Self {
-            tsfn,
-            has_ref: true,
-        }
+        Self::with_trampoline(cx, trampoline)
```

Comment:
`new` returns a referenced trampoline, but `with_trampoline` expects an unreferenced one. I think the best answer here is to have `with_trampoline` take an already-referenced trampoline, which is consistent with fresh ones and not hard to implement for the shared one. Having it take a not-already-referenced one and immediately referencing it would also work, but would mean extra calls into Node for the fresh case.

Reply:
In that case it should probably be called `with_referenced_trampoline` or something so people don't forget.